shell/platform/linux/fl_text_input_channel.cc

```diff
@@ -0,0 +1,275 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/shell/platform/linux/fl_text_input_channel.h"
+#include "flutter/shell/platform/linux/public/flutter_linux/fl_json_method_codec.h"
+
+struct _FlTextInputChannel {
+  FlMethodChannel parent_instance;
+
+  FlTextInputChannelSetClientHandler set_client_handler;
+  FlTextInputChannelShowHandler show_handler;
+  FlTextInputChannelSetEditingStateHandler set_editing_state_handler;
+  FlTextInputChannelClearClientHandler clear_client_handler;
+  FlTextInputChannelHideHandler hide_handler;
+  gpointer handler_data;
+};
+
+G_DEFINE_TYPE(FlTextInputChannel,
+              fl_text_input_channel,
+              fl_method_channel_get_type())
+
+static void method_call_cb(FlMethodChannel* channel,
+                           const gchar* method,
+                           FlValue* args,
+                           FlMethodChannelResponseHandle* response_handle,
+                           gpointer user_data) {
+  FlTextInputChannel* self = FL_TEXT_INPUT_CHANNEL(user_data);
+
+  if (strcmp(method, "TextInput.setClient") == 0) {
+    if (self->set_client_handler != nullptr)
+      self->set_client_handler(self, 0, "", self->handler_data);
+    fl_method_channel_respond(channel, response_handle, nullptr, nullptr);
+  } else if (strcmp(method, "TextInput.show") == 0) {
+    if (self->show_handler != nullptr)
+      self->show_handler(self, self->handler_data);
+    fl_method_channel_respond(channel, response_handle, nullptr, nullptr);
+  } else if (strcmp(method, "TextInput.setEditableSizeAndTransform") == 0) {
```

Comment:
IIRC at least this one is web-only (maybe the next two as well), so I wouldn't include it explicitly, and just let it go into the `else`.

Reply:
I think we need this for input method popups - we'll need to make an X window over this area so they show in the appropriate place.